src/bignum.nr

```diff
@@ -392,6 +393,24 @@ where
     }
 }
 
+impl<let N: u32, let MOD_BITS: u32, Params> std::cmp::Ord for BigNum<N, MOD_BITS, Params>
+where
+    Params: BigNumParamsGetter<N, MOD_BITS>,
+{
+    fn cmp(self, other: Self) -> Ordering {
+        let comp_result: bool = gt::<_, MOD_BITS>(self.limbs, other.limbs);
+        if self.limbs == other.limbs {
+            Ordering::equal()
+        } else if comp_result {
+            // there was an underflow
+            Ordering::greater()
+        } else {
+            // there was no underflow
+            Ordering::less()
+        }
+    }
```

Comment:
Can we push a `cmp` function into `constrained_ops`? We can then avoid the duplicated logic between here and `RuntimeBigNum`.